pdns/dnsdist-lua.cc

```diff
@@ -494,74 +466,60 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
                            }
                          }
 
-                         if (vars.count("checkName")) {
-                           config.checkName = DNSName(boost::get<string>(vars["checkName"]));
-                         }
-
-                         if (vars.count("checkType")) {
-                           config.checkType = boost::get<string>(vars["checkType"]);
-                         }
-
-                         if (vars.count("checkClass")) {
-                           config.checkClass = std::stoi(boost::get<string>(vars["checkClass"]));
+                         if (getOptionalValue<std::string>(vars, "checkName", valueStr) > 0) {
+                           config.checkName = DNSName(valueStr);
                          }
 
-                         if (vars.count("checkFunction")) {
-                           config.checkFunction = boost::get<DownstreamState::checkfunc_t>(vars["checkFunction"]);
-                         }
+                         getOptionalValue<std::string>(vars, "checkType", config.checkType);
 
-                         if (vars.count("checkTimeout")) {
-                           config.checkTimeout = std::stoi(boost::get<string>(vars["checkTimeout"]));
+                         if (getOptionalValue<std::string>(vars, "checkClass", valueStr) > 0) {
+                           config.checkClass = std::stoi(valueStr);
```

Comment:
As done a bit later, do we want to guard potential conversion exceptions for optional values ?

Reply:
I'm also wondering if there is a good reason to not directly `getOptionalValue<int>(vars, "checkClass", config.checkClass)`

Reply:
> I'm also wondering if there is a good reason to not directly `getOptionalValue<int>(vars, "checkClass", config.checkClass)`

Wouldn't that fail because the Lua wrapper used the `string` type for that variable? We could extend the variant to allow integers, but then we need to change all the existing types and make sure the conversion works properly. One drawback that I did not notice before with the new code is that if we pass a string when we expect an integer, it is no longer considered a parsing error and we don't even warn, except at the end to say something like `newServer: Unknown key 'checkClass' given - ignored`, which is confusing: the key exists, but the type is wrong.

Reply:
I pushed a new commit with better handling of numerical values, comments welcome!